src/main/java/org/opensearch/neuralsearch/query/HybridQueryScorer.java

```diff
@@ -194,70 +187,30 @@ public int docID() {
      */
     public float[] hybridScores() throws IOException {
         float[] scores = new float[subScores.length];
-        DisiWrapper topList = subScorersPQ.topList();
-        for (DisiWrapper disiWrapper = topList; disiWrapper != null; disiWrapper = disiWrapper.next) {
+        HybridDisiWrapper topList = (HybridDisiWrapper) subScorersPQ.topList();
```

Comment:
nit: can you check instance type before type case?

Reply:
yes, added the check. My concern was added latency, but seems it's not that big. If we assume 39 ns per `instance of` call, then for my config in experiment 2 it will be 0.3ms, I think that's ok

Reply:
Great... deep-dive again

Reply:
It would have been nice if PriorityQueue uses Generics .

Reply:
we're using not Java PriorityQueue but [Lucene implementation](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/DisiPriorityQueue.java#L30), it's already uses DisiWrapper as type